Add options to enable/disable docs and demos

Originally written by Anthony Viallard
<viallard@syscom-instruments.com>.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
[Vincent: adapt patch to 5.0.5 version]
Signed-off-by: Vicente Olivert Riera <Vincent.Riera@imgtec.com>
[Michael: adapt patch to 5.4.0 version]
Signed-off-by: Michael Fischer <mf@go-sys.de>
[Michael: adapt patch to 6.0.0 version]
Signed-off-by: Michael Fischer <mf@go-sys.de>

--- a/configure.ac	2023-12-19 00:28:58.000000000 +0100
+++ b/configure.ac	2024-05-13 15:41:18.313920528 +0200
@@ -715,6 +715,32 @@
 
 dnl Process rest of with and enable options
 
+dnl Whether we want to build the doc files
+dnl Does nothing here, is passed on to the doc subdir
+AC_ARG_ENABLE(doc,dnl
+[  --disable-doc    do not build doc files])
+dnl build doc files
+if test "$enable_doc" != no; then
+  AC_CONFIG_SUBDIRS(docs)
+  DOCSUBDIR=docs
+else
+  DOCSUBDIR=
+fi
+AC_SUBST(DOCSUBDIR)
+
+dnl Whether we want to build the demo files
+dnl Does nothing here, is passed on to the demo subdir
+AC_ARG_ENABLE(demo,dnl
+[  --disable-demo    do not build demo files])
+dnl build demo files
+if test "$enable_demo" != no; then
+  AC_CONFIG_SUBDIRS(demo)
+  DEMOSUBDIR=demo
+else
+  DEMOSUBDIR=
+fi
+AC_SUBST(DEMOSUBDIR)
+
 dnl without-extra-coordinate may leave a hole in "struct coordinate"
 AC_ARG_WITH(extra-coordinate,dnl
 [  --without-extra-coordinate    leave possible hole in struct coordinate],,)
--- a/Makefile.am	2023-12-19 00:28:58.000000000 +0100
+++ b/Makefile.am	2024-05-13 15:54:21.189906680 +0200
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in -*-Makefile-*-
 AUTOMAKE_OPTIONS = foreign
 
-SUBDIRS = config m4 term src docs man demo share
+SUBDIRS = config m4 term src $(DOCSUBDIR) man $(DEMOSUBDIR) share
 
 EXTRA_DIST = BUGS Copyright FAQ.pdf INSTALL INSTALL.gnu \
 PATCHLEVEL PGPKEYS README RELEASE_NOTES \
